controllers/node_controller.go

```diff
@@ -111,9 +110,11 @@ func (r *nodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (resul
 			return ctrl.Result{}, fmt.Errorf("failed to create new nodeconfig: %w", err)
 		}
 
+		r.log.Info("rebooting instance")
 		if err := nc.SafeReboot(ctx, node); err != nil {
 			return ctrl.Result{}, fmt.Errorf("full instance reboot failed: %w", err)
 		}
+		r.log.V(1).Info("successful reboot", "instance", instanceInfo.Address)
```

Comment:
You can move this log also to `RebootAndReinitialize()` right after we reinitialize SSH.